pkg/autodiscovery/scheduler/meta.go

```diff
@@ -6,12 +6,15 @@
 package scheduler
 
 import (
+	"sync"
+
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // MetaScheduler is a scheduler dispatching to all its registered schedulers
 type MetaScheduler struct {
+	m                sync.Mutex
```

Comment:
We could probably avoid naming it so we can just do `.Lock()` directly on the struct

Reply:
While i agree this is cleaner, all other mutexes in the autodiscovery package are named and private, so I went with consistency.
Would externally locking the struct help to improve readability/performance?